base/src/main/java/io/spine/code/proto/CamelCase.java

```diff
@@ -52,4 +52,35 @@ public static String convert(UnderscoredName name) {
 
         return builder.toString();
     }
+
+    /**
+     * Converts an underscored name to {@code CamelCase} string in the same manner
+     * as Protobuf compiler does.
+     *
+     * <p>Unlike {@link #convert(UnderscoredName)}, capitalizes a letter going after a number,
+     * e.g. {@code p2p} to {@code P2P}.
+     *
+     * @see <a href="https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/compiler/java/java_helpers.cc#L161">
+     *         Protoc Camel Case</a>
+     */
+    @SuppressWarnings("MethodWithMultipleLoops" /* Keep specific logic inside one method. */ )
+    public static String protocStyled(UnderscoredName name) {
+        Iterator<String> iterator = name.words()
+                                        .iterator();
+        StringBuilder builder = new StringBuilder(name.value()
+                                                      .length());
+        while (iterator.hasNext()) {
+            String word = iterator.next();
+            boolean capitalizeNext = true;
+            for (char sourceChar : word.toCharArray()) {
+                char charToAppend = capitalizeNext
+                                    ? Character.toUpperCase(sourceChar)
+                                    : sourceChar;
```

Comment:
Please add a test on what happens if a field has a capital letter in the middle of a word. For example, `type_URL`. 

This is a rare case, since it does not adhere to the Protobuf naming convention, but still, we better check it.